sources/editor/Stride.Core.Assets.Editor/Quantum/NodePresenters/Commands/AddPrimitiveKeyCommand.cs

```diff
@@ -94,7 +99,25 @@ private static object CreateInstance(Type type)
             return ObjectFactoryRegistry.NewInstance(type);
         }
 
-        internal static NodeIndex GenerateStringKey(object dictionary, ITypeDescriptor descriptor, string baseValue)
+        internal static NodeIndex? GenerateGenericKey(object dictionary, ITypeDescriptor descriptor, string baseValue)
+        {
+            // TODO: use a dialog service and popup a message when the given key is invalid
+            DictionaryDescriptor dictionaryDescriptor = descriptor as DictionaryDescriptor;
+            Type keyType = dictionaryDescriptor.KeyType;
+            object key = keyType.Default();
+
+            if (!string.IsNullOrWhiteSpace(baseValue))
+                key = keyType.GetMethod("Parse", new Type[] { typeof(string) }).Invoke(null, new string[] { baseValue });
```

Comment:
The ``GetMethod("Parse", ...`` here and in other cases can be replaced by ``System.ComponentModel.TypeDescriptor.GetConverter(keyType).ConvertTo(baseValue , typeof(string))`` (and other functions of that type like ``CanConvertTo``, etc.) 
**BUT** ``NodeIndex`` supports ``object`` and so does ``DictionaryDescriptor`` for its keys. Is there any reason to use strings instead of the actual object ?

Reply:
Yes, I solve the invalid user problem by deleting my old repository and forking new one.

This command here I just find it's used in dictionary with string key and enum key. My purpose is to fix the number key dictionary, and support other value types as key as much as posible, so I just consider string as key here.
And as a text input ui triggered command, I'd like see it like a string based command function to construct dictionary key. Maybe it can extend to much more types, but not in this merge request.

Reply:
> Maybe it can extend to much more types, but not in this merge request.

Why not, the changes required to introduce this don't seem significantly harder to implement than what you already have to do with replacing reflection ``Parse`` with the ``TypeDescriptor.GetConverter`` stuff. 

If you don't want to dedicate any more time on this PR I understand but I don't think this PR should be merged in this state with it being so close to potentially support any key type.

Reply:
Yeah, I have tried it and it's really a great idea! Thanks for the advice!
I have changed my code to support more generic type keys, with some annoying can't edit bug fixed.